framework/src/common/errors.rs

```diff
@@ -1,98 +1,74 @@
-        BadQueue {
-            description("Invalid queue request")
-            display("Invalid queue request")
-        }
-        CannotSend {
-            description("Cannot send data out port")
-            display("Cannot send data out port")
-        }
-        BadDev(dev: String) {
-            description("Cannot find device")
-            display("Cannot find device: {}", dev)
-        }
-        BadVdev(vdev: String) {
-            description("Bad vdev specification")
-            display("Bad vdev specification: {}", vdev)
-        }
-        BadTxQueue(port: i32, queue: i32) {
-            description("Bad TX queue")
-            display("Bad TX queue {} for port {}", queue, port)
-        }
-        BadRxQueue(port: i32, queue: i32) {
-            description("Bad RX queue")
-            display("Bad RX queue {} for port {}", queue, port)
-        }
-        BadOffset(offset: usize) {
-            description("Attempt to access bad packet offset")
-            display("Attempt to access bad packet offset {}", offset)
-        }
-
-        MetadataTooLarge {
-            description("Metadata is too large")
-            display("Metadata is too large")
-        }
-
-        RingAllocationFailure {
-            description("Could not allocate ring")
-            display("Could not allocate ring")
-        }
-
-        InvalidRingSize(size: usize) {
-            description("Bad ring size, must be power of 2")
-            display("Bad ring size {}, must be a power of 2", size)
-        }
-
-        RingDuplicationFailure {
-            description("Address of second copy of ring does not match expected address")
-            display("Address of second copy of ring does not match expected address")
-        }
-
-        ConfigurationError(description: String) {
-            description("Configuration error")
-            display("Configuration error: {}", description)
-        }
-
-        NoRunningSchedulerOnCore(core: i32) {
-            description("No scheduler running on core")
-            display("No scheduler running on core {}", core)
-        }
-
-        FailedToInsertHeader {
-            description("Failed to insert header into packet")
-            display("Failed to insert header into packet")
-        }
-
-        FailedToSwapHeader(new_header: String) {
-            description("Failed to swap-in new header in packet")
-                display("Failed to swap-in new header - {} - in packet", new_header)
-        }
-
-        FailedToRemoveHeader {
-            description("Failed to remove header from packet")
-                display("Failed to remove header from packet")
-        }
-
-        FailedToParseMacAddress(s: String) {
-            description("Failed to parse MAC address")
-                display("Failed to parse MAC address: '{}'", s)
-        }
-    }
-
-    foreign_links {
-        Io(::std::io::Error);
-    }
+use failure::{Error, Fail};
+
+pub type Result<T> = ::std::result::Result<T, Error>;
```

Comment:
If I'm reading this correctly, this is why we have to do an additional conversion/wrap with `into` everywhere

```
Err(NetBricksError::ConfigurationError("foo).into())
```

I think instead if network functions just bubble up `NetBricksError` instead, we can avoid that.

```
pub type Result<T> = ::std::result::Result<T, NetBricksError>;
```

Reply:
After some discussion with @zeeshanlakhani, we concluded that there's no way to avoid `.into()` calls in general from a network function. Whether we convert everything to `failure::Error` or `NetBricksError` doesn't really make any difference. Keeping it `failure::Error` is likely the simpler design.